libvcx/src/lib.rs

```diff
@@ -287,22 +319,39 @@ mod tests {
         info!("test_basic_revocation :: Going to seng proof request with attributes {}", &requested_attrs_string);
         let proof_handle_verifier = send_proof_request(alice, &requested_attrs_string, "[]", &interval);
 
-        info!("test_basic_revocation :: Going to create proof");
-        let proof_handle_prover = create_proof(faber);
-        info!("test_basic_revocation :: retrieving matching credentials");
+        _prover_select_credentials_and_send_proof(faber);
 
-        let retrieved_credentials = disclosed_proof::retrieve_credentials(proof_handle_prover).unwrap();
-        info!("test_basic_revocation :: prover :: based on proof, retrieved credentials: {}", &retrieved_credentials);
+        info!("test_basic_revocation :: verifier :: going to verify proof");
+        set_institution();
+        proof::update_state(proof_handle_verifier, None, None).unwrap();
+        assert_eq!(proof::get_proof_state(proof_handle_verifier).unwrap(), ProofStateType::ProofInvalid as u32);
+    }
 
-        let selected_credentials_value = retrieved_to_selected_credentials_simple(&retrieved_credentials, true);
-        let selected_credentials_str = serde_json::to_string(&selected_credentials_value).unwrap();
-        info!("test_basic_revocation :: prover :: retrieved credential converted to selected: {}", &selected_credentials_str);
-        generate_and_send_proof(proof_handle_prover, faber, &selected_credentials_str);
+    #[cfg(feature = "agency_pool_tests")]
+    #[test]
+    fn test_local_revocation() {
+        let _setup = SetupLibraryAgencyV2ZeroFees::init();
+
+        let institution_did = settings::get_config_value(settings::CONFIG_INSTITUTION_DID).unwrap();
+        let (faber, alice) = ::connection::tests::create_connected_connections();
+        let (schema_id, cred_def_id, rev_reg_id, _cred_def_handle, credential_offer) = _issue_address_credential(faber, alice, &institution_did);
```

Comment:
The fact that the last element is credential handle but it is called `credential_offer` is bit confusing

Reply:
Also in the `_issue_address_credential ` function itself